linkml_runtime/utils/dataclass_extensions_376.py

```diff
@@ -1,30 +1,54 @@
+import sys
 import dataclasses
-#
-# The dataclass library builds a rigid `__init__` function that doesn't allow any unrecognized named parameters
-#
-# The purpose of this extension is to enhance the library to allow additional keyword arguments to passed in
-# and then on to the __post_init__ function that can deal with them accordingly
+import warnings
 
-# Beware that there is no promise that signature of the create function will remain consistent
-loc_fn = dataclasses._create_fn
+# Block unsupported Python versions
+if sys.version_info >= (3, 13):
+    raise RuntimeError(
```

Comment:
if i read the discussion in slack correctly, we want this to *not* fail for existing code right? 

The monkeypatch that was removed here is functionally equivalent to the standard dataclass behavior, it's just a matter of when the error for extra parameters is raised. so we could just gate the whole module in this check, and in the python 3.13 branch create some dummy function that does nothing but allows the import to proceed, and then we both preserve existing functionality and allow python 3.13 to work

Reply:
removing this since it's outdated!